test-foundry/L2USDCCustomGateway.t.sol

```diff
@@ -0,0 +1,269 @@
+        vm.prank(sender);
+        l2USDCGateway.outboundTransfer(l1USDC, receiver, withdrawalAmount, data);
+    }
+
+    function test_outboundTransfer_revert_NotExpectedL1Token() public override {
+        // mock invalid L1 token ref
+        address notl1USDC = makeAddr("notl1USDC");
+        vm.mockCall(address(l2USDC), abi.encodeWithSignature("l1Address()"), abi.encode(notl1USDC));
+
+        vm.expectRevert("NOT_EXPECTED_L1_TOKEN");
+        l2USDCGateway.outboundTransfer(l1USDC, address(101), 200, 0, 0, new bytes(0));
+    }
+
+    function test_outboundTransfer_revert_WithdrawalsPaused() public {
+        // pause withdrawals
+        vm.prank(owner);
+        l2USDCGateway.pauseWithdrawals();
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                L2USDCCustomGateway.L2USDCCustomGateway_WithdrawalsPaused.selector
+            )
+        );
+        l2USDCGateway.outboundTransfer(l1USDC, receiver, 200, 0, 0, new bytes(0));
+    }
+
+    function test_pauseWithdrawals() public {
+        assertEq(l2USDCGateway.withdrawalsPaused(), false, "Invalid initial state");
+
+        // events
+        vm.expectEmit(true, true, true, true);
+        emit WithdrawalsPaused();
+
+        // pause withdrawals
+        vm.prank(owner);
+        l2USDCGateway.pauseWithdrawals();
+
+        // checks
+        assertEq(l2USDCGateway.withdrawalsPaused(), true, "Invalid initial state");
+    }
+
+    function test_pauseWithdrawals_revert_WithdrawalsAlreadyPaused() public {
+        vm.startPrank(owner);
+        l2USDCGateway.pauseWithdrawals();
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                L2USDCCustomGateway.L2USDCCustomGateway_WithdrawalsAlreadyPaused.selector
+            )
+        );
+        l2USDCGateway.pauseWithdrawals();
+    }
+
+    function test_pauseWithdrawals_revert_NotOwner() public {
+        vm.expectRevert(
+            abi.encodeWithSelector(L2USDCCustomGateway.L2USDCCustomGateway_NotOwner.selector)
+        );
+        l2USDCGateway.pauseWithdrawals();
+    }
+
+    function test_setOwner() public {
+        address newOwner = makeAddr("new-owner");
+        vm.prank(owner);
+        l2USDCGateway.setOwner(newOwner);
+
+        assertEq(l2USDCGateway.owner(), newOwner, "Invalid owner");
+    }
+
+    function test_setOwner_revert_InvalidOwner() public {
+        vm.expectRevert(
+            abi.encodeWithSelector(L2USDCCustomGateway.L2USDCCustomGateway_InvalidOwner.selector)
+        );
+        vm.prank(owner);
+        l2USDCGateway.setOwner(address(0));
+    }
+
+    function test_setOwner_revert_NotOwner() public {
+        vm.expectRevert(
+            abi.encodeWithSelector(L2USDCCustomGateway.L2USDCCustomGateway_NotOwner.selector)
+        );
+        l2USDCGateway.setOwner(owner);
+    }
+
+    ////
+    // Event declarations
+    ////
+    event WithdrawalsPaused();
+}
+
+contract L2USDC is L2GatewayToken {
```

Comment:
this should be a separate prod contract, not in test

Reply:
Removed in separate PR - https://github.com/OffchainLabs/token-bridge-contracts/pull/88